tests/test_srv6.py

```diff
@@ -0,0 +1,260 @@
+        tbl = swsscommon.Table(self.adb.db_connection, "ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY")
+        (status, fvs) = tbl.get(route_key)
+        assert status == True
+        for fv in fvs:
+            if fv[0] == "SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID":
+                assert fv[1] == nexthop_id
+
+        # check ASIC SAI_OBJECT_TYPE_NEXT_HOP database
+        tbl = swsscommon.Table(self.adb.db_connection, "ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP")
+        (status, fvs) = tbl.get(nexthop_id)
+        assert status == True
+        for fv in fvs:
+            if fv[0] == "SAI_NEXT_HOP_ATTR_SRV6_SIDLIST_ID":
+                assert fv[1] == sidlist_id
+            elif fv[0] == "SAI_NEXT_HOP_ATTR_TUNNEL_ID":
+                assert fv[1] == tunnel_id
+
+        # check ASIC SAI_OBJECT_TYPE_TUNNEL database
+        tbl = swsscommon.Table(self.adb.db_connection, "ASIC_STATE:SAI_OBJECT_TYPE_TUNNEL")
+        (status, fvs) = tbl.get(tunnel_id)
+        assert status == True
+        for fv in fvs:
+            if fv[0] == "SAI_TUNNEL_ATTR_TYPE":
+                assert fv[1] == "SAI_TUNNEL_TYPE_SRV6"
+            elif fv[0] == "SAI_TUNNEL_ATTR_ENCAP_SRC_IP":
+                assert fv[1] == "1001:2000::1"
+
+
+        # create 2nd seg lists
+        self.create_sidlist('seg2', 'baba:2002:10::,baba:2002:20::')
+        # create 3rd seg lists
+        self.create_sidlist('seg3', 'baba:2003:10::,baba:2003:20::')
+
+        # create 2nd v4 route with single sidlists
+        self.create_srv6_route('20.20.20.21/32','seg2','1001:2000::1')
+        # create 3rd v4 route with single sidlists
+        self.create_srv6_route('20.20.20.22/32','seg3','1001:2000::1')
+
+        # remove routes
+        self.remove_srv6_route('20.20.20.20/32')
+        self.check_deleted_route_entries('20.20.20.20/32')
+        self.remove_srv6_route('20.20.20.21/32')
+        self.check_deleted_route_entries('20.20.20.21/32')
+        self.remove_srv6_route('20.20.20.22/32')
+        self.check_deleted_route_entries('20.20.20.22/32')
+
+        # remove sid lists
+        self.remove_sidlist('seg1')
+        self.remove_sidlist('seg2')
+        self.remove_sidlist('seg3')
+
+        # remove neighbors
+        self.remove_neighbor("Ethernet104", "baba:2001:10::")
+        self.remove_neighbor("Ethernet112", "baba:2002:10::")
+        self.remove_neighbor("Ethernet120", "baba:2003:10::")
+
+        # check if asic db entries are all restored
+        assert tunnel_entries == get_exist_entries(self.adb.db_connection, "ASIC_STATE:SAI_OBJECT_TYPE_TUNNEL")
+        assert nexthop_entries == get_exist_entries(self.adb.db_connection, "ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP")
+        assert route_entries == get_exist_entries(self.adb.db_connection, "ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY")
+
+# Add Dummy always-pass test at end as workaroud
+# for issue when Flaky fail on final test it invokes module tear-down before retrying
+def test_nonflaky_dummy():
+    pass
```

Comment:
should we add some negative test cases? e.g. invalid SRv6 sids setting?

Reply:
@caizhenghui-juniper Will add the negative tests in the next PR.